tensorpipe/test/channel/channel_test.h

```diff
@@ -76,14 +74,22 @@ class DataWrapper<tensorpipe::CudaBuffer> {
         stream_(other.stream_) {
     other.cudaPtr_ = nullptr;
     other.length_ = 0;
+    other.stream_ = cudaStreamDefault;
+  }
+
+  DataWrapper& operator=(DataWrapper&& other) {
+    cudaPtr_ = other.cudaPtr_;
+    length_ = other.length_;
+    stream_ = other.stream_;
+    other.cudaPtr_ = nullptr;
+    other.length_ = 0;
+    other.stream_ = cudaStreamDefault;
+
+    return *this;
   }
 
   tensorpipe::CudaBuffer buffer() const {
-    return tensorpipe::CudaBuffer{
-        .ptr = cudaPtr_,
-        .length = length_,
-        .stream = stream_,
-    };
+    return tensorpipe::CudaBuffer{cudaPtr_, length_, stream_};
   }
 
   std::vector<uint8_t> unwrap() {
```

Comment:
Return by const-ref here too?

Reply:
Returning a const ref here is UB, as the vector is stack-allocated. Moreover, copy-elision should prevent actually copying the object.

Reply:
Sorry, you're right.